module_utils/kafka_manager.py

```diff
@@ -183,6 +191,60 @@ def delete_topics(self, topics, timeout=None):
                     )
                 )
 
+    def get_coordinator_for_consumer_group(self, consumer_group=None):
+
+        # Identify which broker is coordinating this consumer group
+        response = self.send_request_and_get_response(
+                 GroupCoordinatorRequest_v0(consumer_group)
+                 )
+        if response.error_code != self.SUCCESS_CODE:
+            raise KafkaManagerError(
+                'Error getting coordinator for consumer group %s.' % (
+                    consumer_group)
+            )
+        return response.coordinator_id
+
+    def delete_group_offset(self, group_id, topics, check_mode):
+
+        consumer_coordinator = self.get_coordinator_for_consumer_group(
+            group_id)
+
+        consumed_offsets = self.send_request_and_get_response(
+            OffsetFetchRequest_v2(group_id, None), consumer_coordinator)
+
+        consumed_topics = [e[0] for e in consumed_offsets.topics]  # topic name
+
+        changed = False
+
+        topics_partitions = []
+        for topic in topics:
+            topic_name = topic['name']
+            partitons = topic['partitions']
+            if(partitons is None):
```

Comment:
just wondering if the `partitions` key is always there and if we may need to be careful using `topic.get('partitions')` instead? also should we check for `if not partitons:` instead, since an user can provide an empty list of partitions meaning he actually wants all partitions?

Reply:
done